Mathlib/MeasureTheory/Function/EssSup.lean

```diff
@@ -68,6 +67,38 @@ theorem essSup_const (c : β) (hμ : μ ≠ 0) : essSup (fun _ : α => c) μ = c
 theorem essInf_const (c : β) (hμ : μ ≠ 0) : essInf (fun _ : α => c) μ = c :=
   have := NeZero.mk hμ; essInf_const' _
 
+section SMul
+variable {R : Type*} [Zero R] [SMulWithZero R ℝ≥0∞] [IsScalarTower R ℝ≥0∞ ℝ≥0∞]
+  [NoZeroSMulDivisors R ℝ≥0∞] {c : R}
+
+@[simp]
+lemma essSup_smul_measure (hc : c ≠ 0) (f : α → β) : essSup f (c • μ) = essSup f μ := by
+  simp_rw [essSup, Measure.ae_smul_measure_eq hc]
+
+end SMul
+
+variable [Nonempty α]
+
+lemma essSup_eq_ciSup (hμ : ∀ a, μ {a} ≠ 0) (hf : BddAbove (Set.range f)) :
+    essSup f μ = ⨆ a, f a := by rw [essSup, ae_eq_top.2 hμ, limsup_top_eq_ciSup hf]
+
+lemma essInf_eq_ciInf (hμ : ∀ a, μ {a} ≠ 0) (hf : BddBelow (Set.range f)) :
+    essInf f μ = ⨅ a, f a := by rw [essInf, ae_eq_top.2 hμ, liminf_top_eq_ciInf hf]
+
+variable [MeasurableSingletonClass α]
+
+@[simp] lemma essSup_count_eq_ciSup (hf : BddAbove (Set.range f)) :
+    essSup f .count = ⨆ a, f a := essSup_eq_ciSup (by simp) hf
+
+@[simp] lemma essInf_count_eq_ciInf (hf : BddBelow (Set.range f)) :
+    essInf f .count = ⨅ a, f a := essInf_eq_ciInf (by simp) hf
+
+@[simp] lemma essSup_cond_count_eq_ciSup [Finite α] (hf : BddAbove (Set.range f)) :
+    essSup f .count[|.univ] = ⨆ a, f a := essSup_eq_ciSup (by simp [cond_apply, Set.finite_univ]) hf
```

Comment:
I was looking for the reason you import the file about conditional probability, and apparently it is to state two lemmas about `Measure.count[|.univ]`, which is just a weird spelling for the uniform measure on the type.
I would like to avoid generalizing the use of `.count[|.univ]` to refer to the uniform distribution on a finite type. We should introduce a better name for that.

My proposal: remove those two lemmas and the import, and I merge the PR. Then we separately introduce a definition with a good name for the uniform distribution and reintroduce those lemmas later.
Related question if you look into that: do we want to keep the definition [ProbabilityTheory.condCount](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Probability/CondCount.html#ProbabilityTheory.condCount) ? Or do we want to use `cond[|s]`?

Reply:
I was following @urkud's suggestion from [here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/average.20and.20cond/near/465016842). I will remove the lemmas for now, but I really need some version of this for APAP